pkg/conversion/scheme.go

```diff
@@ -19,22 +19,19 @@ package conversion
 import (
 	"fmt"
 	"reflect"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
 // Scheme defines an entire encoding and decoding scheme.
 type Scheme struct {
 	// versionMap allows one to figure out the go type of an object with
 	// the given version and name.
-	versionMap map[string]map[string]reflect.Type
-
-	// typeToVersion allows one to figure out the version for a given go object.
-	// The reflect.Type we index by should *not* be a pointer. If the same type
-	// is registered for multiple versions, the last one wins.
-	typeToVersion map[reflect.Type]string
+	gvkToType map[unversioned.GroupVersionKind]reflect.Type
```

Comment:
I would prefer we avoid acronyms if possible:  groupVersionKindToType

If unversioned.GroupVersionKind is too much of a mouthful, we should shorten that into more abstract term ;-)